src/execution/__tests__/executor-test.js

```diff
@@ -306,11 +307,65 @@ describe('Execute: Handles basic execution tasks', () => {
     const field = operation.selectionSet.selections[0];
     expect(resolvedInfo).to.deep.include({
       fieldNodes: [field],
-      path: { prev: undefined, key: 'result' },
+      path: { prev: undefined, key: 'result', parentType: 'Test' },
       variableValues: { var: 'abc' },
     });
   });
 
+  it('populates path correctly with complex types', () => {
+    let path;
+    const a = new GraphQLObjectType({
+      name: 'A',
+      fields: {
+        test: {
+          type: GraphQLString,
+          resolve(_val, _args, _ctx, info) {
+            path = info.path;
+          },
+        },
+      },
+    });
+    const testUnion = new GraphQLUnionType({
+      name: 'TestUnion',
+      types: [a],
+      resolveType() {
+        return a;
+      },
+    });
+    const testType = new GraphQLObjectType({
+      name: 'Test',
+      fields: {
+        test: {
+          type: new GraphQLNonNull(
+            new GraphQLList(new GraphQLNonNull(testUnion)),
+          ),
+          resolve(_val, _args, _ctx, _info) {
```

Comment:
Since you have `resolveType` you don't really need this resolver.
Let's keep tests minimal.

Reply:
If we don't have a resolver here, then GraphQL will surely throw an error since the `test` field will not return anything? I could keep the resolver but change it to `[{}]` if you like?

Reply:
Oh; we could use rootValue. I'll do that :+1: